### ISSUE TITLE
Allow grenade pin pull during LCA

### ISSUE THREAD

**Issue by Raptor007** [Raptor007/aq2-tng#143]
Currently if you're playing Bandolier and want to use your grenade right after ACTION! You're essentially useless in the beginning. Since you cannot pull the pin and throw until the announcer yells ACTION! 

To make the grenade(s) a bit more tactical/useful it would be cool if it was possible to pull the pin a bit earlier like when the announcer yells LIGHTS...

Seeing a grenade coming at you or on the floor just at the ACTION! Announcement would probably make players stop and wait until it has detonated. This could be a gamechanger for that specific round!

_Originally posted by @MVarland in https://github.com/Raptor007/aq2-tng/issues/136#issuecomment-1271787717_

**Comment by Raptor007** [comment]
I like the idea of allowing the pin to be pulled at any time during LCA but only allowing the throw after Action.

The only problematic edge case I can think of is changing weapons after pulling the grenade pin.  Normally that drops a live grenade at your feet, but this shouldn't be allowed before Action.

**Comment by darkshade9** [comment]
Any way to detect if we're in LCA and if ent has grenade equipped, and grenade animation (pin pulled) is after the pin pull frame, make it unable for player to switch weapon?  Sort of a 'point of no return' once pin is pulled?

**Comment by Raptor007** [comment]
Yeah, there could be a point of no return if we want to be strict about pin pulling.  I think a better solution would be if you weapon switch during LCA, it just doesn't drop the live grenade, essentially "putting the pin back in" instead.  That would be a little easier to code and I think more friendly to players.  (Or could have a cvar and let server admins decide, but that's the most work and maybe not worth it for this little change.)
